### PR TITLE
fixes 98, fixes 101

### PR DESCRIPTION
**Pull request by jvanhoefer**

### CODE REVIEW

File: yaml2sbml/yaml2sbml.py [hunk 1]
```diff
@@ -65,9 +93,19 @@ def _parse_yaml_dict(yaml_dict) -> str:
         raise SystemExit('Could not create SBMLDocument object')
 
     model = document.createModel()
+
+    # remove file extension
+    if model_name.endswith('.xml') or model_name.endswith('.sbml'):
+        model_name = model_name[:-4]
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:99]: .sbml are 5 ...

File: yaml2sbml/yaml2sbml.py [hunk 2]
```diff
@@ -372,7 +418,9 @@ def _create_rate_rule(model: sbml.Model, species: str, formula: str):
     r.setMath(math_ast)
 
 
-def _read_observables_block(model: sbml.Model, observable_list: list):
+def _read_observables_block(model: sbml.Model,
+                            observable_list: list,
+                            observables_as_assignments: bool):
     """
     Read an process the observables block in the YAML file.
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:425]:
```suggestion
    Read and process the observables block in the YAML file.
```

File: yaml2sbml/yaml2sbml.py [hunk 3]
```diff
@@ -383,10 +431,18 @@ def _read_observables_block(model: sbml.Model, observable_list: list):
...
+    """
+    if observables_as_assignments:
+        for observable_def in observable_list:
+            _create_assignment(model,
+                               f'observable_{observable_def["observableId"]}',
+                               observable_def['observableFormula'])
+    else:
+        warnings.warn(
+            'Observables are not represented in the SBML and therefore only '
+            'have an effect on the output when called via yaml2PEtab')
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:445]:
```suggestion
            'have an effect on the output when called via yaml2petab')
```

File: tests/test_CLI.py [hunk 4]
```diff
@@ -15,13 +15,20 @@ def test_yaml2sbml_cli(script_runner):
...
     with open(sbml_dir, 'r') as f_in:
         sbml_from_cli = f_in.read()
 
-    if not (sbml_from_cli == sbml_from_python):
-        raise AssertionError('SBML from CLI and python interface differ.')
+    for line_cli, line_python in zip(sbml_from_cli.split('\n'),
+                                     sbml_from_python.split('\n')):
+
+        # the line containing name & id will not match...
+        if not (line_cli == line_python or line_cli.startswith(
```
Comment (yannikschaelte) [inline tests/test_CLI.py:27]: why is the model id only contained in the cli version but not the python version? should it not be the same calls?

File: yaml2sbml/YamlModel.py [hunk 5]
```diff
@@ -109,14 +109,18 @@ def write_to_sbml(self,
             raise ValueError('sbml_dir should contain path to the sbml '
                              'and hence end with .xml or .sbml')
 
+        # model name = sbml name without file extension
+        model_name = os.path.splitext(sbml_dir)[0][:-4]
```
Comment (yannikschaelte) [inline yaml2sbml/YamlModel.py:113]: why -4? is this always 4 letters?

**Review by yannikschaelte — APPROVED** [review]
looks good, some comments

File: yaml2sbml/YamlModel.py (hunk 5 above)
Comment (dilpath) [reply yaml2sbml/YamlModel.py:113]: The following may be sufficient
```python
from pathlib import Path
# The name of the model is the SBML filename without its extension.
model_name = Path(sbml_dir).stem
```

File: yaml2sbml/yaml2sbml.py (hunk 1 above)
Comment (dilpath) [reply yaml2sbml/yaml2sbml.py:99]: As above, using `pathlib.Path` should make this easy

Comment (jvanhoefer) [reply yaml2sbml/yaml2sbml.py:99]: As the common maths joke says: There are three sorts of mathematicians: Those who can count, and those who don't -.- :D Thank you for catching this :+1:

File: yaml2sbml/YamlModel.py (hunk 5 above)
Comment (jvanhoefer) [reply yaml2sbml/YamlModel.py:111]: thank you @dilpath (looking at your name, you of course know path lib :D (sorry, bad pun...))

File: tests/test_CLI.py (hunk 4 above)
Comment (jvanhoefer) [reply tests/test_CLI.py:27]: You are right, naming them both the same will solve the problem. Didn't think of that.
(The problem was, that they had different model names before, so. every line was the same despite the one with the model id. fixed that.)

File: yaml2sbml/yaml2sbml.py [hunk 6]
```diff
@@ -1,39 +1,59 @@
...
 import argparse
 import warnings
+from pathlib import Path
 
 import libsbml as sbml
 import yaml
 from yaml.scanner import ScannerError
 
-from .yaml_validation import _validate_yaml_from_dict
+from yaml_validation import _validate_yaml_from_dict
```
Comment (yannikschaelte) [inline yaml2sbml/yaml2sbml.py:10]: this could cause the problems on GHA?

**Pull request merged**